web-common/src/features/canvas/CanvasDashboardEmbed.svelte

```diff
@@ -1,32 +1,66 @@
 <script lang="ts">
   import CanvasDashboardWrapper from "./CanvasDashboardWrapper.svelte";
-  import { getCanvasStore } from "./state-managers/state-managers";
+  import {
+    getCanvasStoreUnguarded,
+    type CanvasStore,
+  } from "./state-managers/state-managers";
   import StaticCanvasRow from "./StaticCanvasRow.svelte";
   import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
   import Spinner from "../entity-management/Spinner.svelte";
   import { EntityStatus } from "../entity-management/types";
+  import { derived, get, type Readable, type Writable } from "svelte/store";
+  import type { Row } from "./stores/row";
+  import { getEmbedThemeStoreInstance } from "../embeds/embed-theme-store";
+  import { resolveEmbedTheme } from "../embeds/embed-theme-utils";
 
   export let canvasName: string;
   export let navigationEnabled: boolean = true;
 
   $: ({ instanceId } = $runtime);
 
-  $: ({
-    canvasEntity: {
-      components,
-      _rows,
-      firstLoad,
-      _maxWidth,
-      filtersEnabledStore,
-    },
-  } = getCanvasStore(canvasName, instanceId));
-
-  $: filtersEnabled = $filtersEnabledStore;
-  $: maxWidth = $_maxWidth;
-  $: rows = $_rows;
+  let canvasStore: CanvasStore | undefined;
+  let components: CanvasStore["canvasEntity"]["components"];
+  let _rows: Readable<Row[]>;
+  let firstLoad: Readable<boolean>;
+  let _maxWidth: Readable<number>;
+  let filtersEnabledStore: Readable<boolean | undefined>;
+  let themeName: Writable<string | undefined>;
+  let rows: Row[] = [];
+
+  // Look up the canvas store without throwing if it doesn't exist yet.
+  $: canvasStore = getCanvasStoreUnguarded(canvasName, instanceId);
```

Comment:
nit: `CanvasProvider` is supposed to make the store available already, can you add a TODO to debug why we need this?